File: java/sdk/sdk-pay/src/main/java/com/novax/sdk/pay/receipt/ReceiptOrdersRequest.java
package com.novax.sdk.pay.receipt;

import com.novax.sdk.core.request.AbstractApiRequest;
import com.novax.sdk.core.request.HttpMethod;
import com.novax.sdk.core.request.TypeRef;
import com.novax.sdk.pay.model.ReceiptOrderResp;
import lombok.Builder;
import lombok.Getter;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

/** GET /pay/v3/receipt/orders — query receipt orders. */
@Getter
@Builder
public class ReceiptOrdersRequest extends AbstractApiRequest<List<ReceiptOrderResp>> {

    private final String receiptOrderIds;
    private final Integer orderStatus;
    private final Integer status;

    @Override public HttpMethod method() { return HttpMethod.GET; }
    @Override public String path() { return "/pay/v3/receipt/orders"; }

    @Override public Map<String, ?> queryParams() {
        Map<String, Object> q = new LinkedHashMap<>();
        if (receiptOrderIds != null) q.put("receiptOrderIds", receiptOrderIds);
        if (orderStatus != null) q.put("orderStatus", orderStatus);
        if (status != null) q.put("status", status);
        return q;
    }

    @Override public TypeRef<List<ReceiptOrderResp>> responseType() {
        return new TypeRef<>() {};
    }
}

File: python/novax_sdk/pay/receipt/receipt_orders_request.py
from typing import Any, Optional

from novax_sdk.pay.model import ReceiptOrderResp
from novax_sdk.request.api_request import ApiRequest, HttpMethod


class ReceiptOrdersRequest(ApiRequest):
    """GET /pay/v3/receipt/orders"""

    def __init__(self, receipt_order_ids: Optional[str] = None,
                 order_status: Optional[int] = None,
                 status: Optional[int] = None) -> None:
        self._receipt_order_ids = receipt_order_ids
        self._order_status = order_status
        self._status = status

    @property
    def method(self) -> HttpMethod:
        return HttpMethod.GET

    @property
    def path(self) -> str:
        return "/pay/v3/receipt/orders"

    def query_params(self) -> dict[str, Any]:
        return {k: v for k, v in {
            "receiptOrderIds": self._receipt_order_ids,
            "orderStatus": self._order_status,
            "status": self._status,
        }.items() if v is not None}

    @property
    def response_type(self):
        return list[ReceiptOrderResp]
